### ISSUE TITLE
How did you train the flownet with a scale layer?

### ISSUE THREAD

**Issue by WBinke** [XUSean0118/DVSNet#7]
Hi,I have a quession about the flownet.The original flownet doesn't have a scale layer, but in your version it has a scale layer in the last of the flownet. But it doesn't have any  supervising clue to train the scale layer.Could you tell me how to train it? Thanks!

**Comment by XUSean0118** [comment]
Hi, WBinke
I add the scale layer which initialized all 0 when I finetune the pre-train combination model (flow +segmentation).

**Comment by WBinke** [comment]
Thanks for your reply! But I still have some questions about the finetuning part, in your release code I don't see the finetuning part,  is it just like the Deep Feature Flow? Could you release the finetuning part or tell me more details about this part? Thank you very much!

**Comment by WBinke** [comment]
@XUSean0118 Hi，recently I'm working on reimplementing the finetuning part using different segmentation model.Could you tell me more about the finetuning details just like the input size and optimizer and provide the pre-trained flownet model to me? Thanks!

**Comment by XUSean0118** [comment]
@WBinke 
Hi, here is the pre-trained flownet model "https://drive.google.com/drive/folders/1sEXigD4tBjXePQlbsPde45zkBQQ5iD1Z?usp=sharing".

In the fine tuning phase, randomly select a pair of nearby video frames {key frame, current frame}, where 0 ≤ current frame − key frame ≤ 9. In the forward pass, the segmentation network is applied on the
key frame to get the segmented result of the key frame and the flow network estimates the flow field and the scale field  with the input pair {key frame, current frame}. Then, warping the segmented result of key frame to get the segmented result of the current frame. The loss function is the sum of cross-entropy terms for each pixel in the segmented result of the current frame with the ground truth labels (except for unlabeled pixels which are ignored). The gradients of the loss are back-propagated to update the weights by the Adaptive Moment Estimation (Adam) Optimizer.

You can obtain 10 image pairs {key frame, current frame}, where key frame is the 20th frame, from each training snippets of Cityscapes dataset. Then, we randomly crop 1,024×512 from original resolution 2,048×1,024 image pairs. We perform Adam 300K iterations on cropped image pairs with mini-batch size equals 4, where the learning rates are set to 10−4 and 10−5 for the first 150K and the last 150K iterations, respectively.

**Comment by XUSean0118** [comment]
@WBinke
Hi, here is the pre-trained flownet model "https://drive.google.com/drive/folders/1sEXigD4tBjXePQlbsPde45zkBQQ5iD1Z?usp=sharing".

In the fine tuning phase, randomly select a pair of nearby video frames {key frame, current frame}, where 0 ≤ current frame − key frame ≤ 9. In the forward pass, the segmentation network is applied on the
key frame to get the segmented result of the key frame and the flow network estimates the flow field and the scale field with the input pair {key frame, current frame}. Then, warping the segmented result of key frame to get the segmented result of the current frame. The loss function is the sum of cross-entropy terms for each pixel in the segmented result of the current frame with the ground truth labels (except for unlabeled pixels which are ignored). The gradients of the loss are back-propagated to update the weights by the Adaptive Moment Estimation (Adam) Optimizer.

You can obtain 10 image pairs {key frame, current frame}, where key frame is the 20th frame, from each training snippets of Cityscapes dataset. Then, randomly crop 1,024×512 from original resolution 2,048×1,024 image pairs. I perform Adam 300K iterations on cropped image pairs with mini-batch size equals 4, where the learning rates are set to 10−4 and 10−5 for the first 150K and the last 150K iterations, respectively.

**Comment by WBinke** [comment]
@XUSean0118 Thanks for your reply!But I still have two quessions.First,in your reply,the key frame is the 20th frame,but I think the current frame should be the 20th frame because it has the gt lables for training flownet.And second,I want to ask the specific parameter setting about the Adam Optimizer.

**Comment by XUSean0118** [comment]
Hi, @WBinke
You are right, the current frame should be the 20th frame. And beta1=0.9,beta2=0.999 for the Adam Optimizer.
